django/db/backends/oracle/creation.py

```diff
@@ -40,6 +40,11 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
                 if keepdb:
                     return
                 sys.stderr.write("Got an error creating the test database: %s\n" % e)
+                if 'ORA-01031' in str(e):
+                    sys.stderr.write(
+                        "It appears the main user has insufficient privileges "
+                        "to create test database.\n")
+                    sys.exit(2)
```

Comment:
I think this check should go before `if keepdb:` in line 40; with the current code, if the database doesn't exist and `keepdb` was specified, and anything went wrong, creation will be silently skipped and this is incorrect.